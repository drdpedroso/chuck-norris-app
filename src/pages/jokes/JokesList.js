import React from 'react'
import PropTypes from 'prop-types'
import JokeListItem from './jokesList/JokesListItem'

export const JokesList = ({ jokes = [], onChange = () => {}, list }) => {
  const renderListItem = items => {
    return Object.keys(items).map(key => {
      const {id, favorite, joke} = items[key]
      return (<JokeListItem key={id} id={id} favorite={favorite} joke={joke} onChange={onChange} list={list}/>)
    })
  }

  return (
    <React.Fragment>
      {renderListItem(jokes)}
    </React.Fragment>
  )
}


JokesList.propTypes = {
  jokes: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    joke: PropTypes.string,
    categories: PropTypes.arrayOf(PropTypes.string),
  })).isRequired,
}

JokesList.defaultProps = {
  jokes: [{
    id: 0,
    joke: 'Just a filler joke',
    categories: [],
  }],
}

export default JokesList
