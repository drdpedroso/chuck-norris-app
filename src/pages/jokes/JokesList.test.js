import React from 'react'
import renderer from 'react-test-renderer'
import { Joke } from 'services/tests/factories'
import JokesList from './JokesList'

const favoritedJokes = [
  Joke.build({ favorite: true }),
  Joke.build({ favorite: true }),
  Joke.build({ favorite: true }),
]

const regularJokes = [
  Joke.build({ favorite: false }),
  Joke.build({ favorite: false }),
  Joke.build({ favorite: false }),
]

describe('JokesList', () => {
  it('should render correctly', () => {
    const component = renderer.create(<JokesList jokes={favoritedJokes} />)
    expect(component).toMatchSnapshot()
  })
  it('should render correctly', () => {
    const component = renderer.create(<JokesList jokes={regularJokes} />)
    expect(component).toMatchSnapshot()
  })
})
