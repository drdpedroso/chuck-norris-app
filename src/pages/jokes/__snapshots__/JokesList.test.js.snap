// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JokesList should render correctly 1`] = `
Array [
  <div>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Chuck Norris crossed the road. No one has ever dared question his motives.
  </div>,
  <div>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Chuck Norris crossed the road. No one has ever dared question his motives.
  </div>,
  <div>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Chuck Norris crossed the road. No one has ever dared question his motives.
  </div>,
]
`;

exports[`JokesList should render correctly 2`] = `
Array [
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Chuck Norris crossed the road. No one has ever dared question his motives.
  </div>,
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Chuck Norris crossed the road. No one has ever dared question his motives.
  </div>,
  <div>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    Chuck Norris crossed the road. No one has ever dared question his motives.
  </div>,
]
`;
